kroxylicious-authorizer/src/main/java/io/kroxylicious/authorizer/service/ResourceType.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.authorizer.service;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * An Operation is an {@code enum} of the possible operations on a resource of a particular type.
+ * We use a one-enum-per-resource-type pattern so that the {@link Class} of an operation identifies the resource type.
+ * @param <S>
+ */
+public interface ResourceType<S extends Enum<S> & ResourceType<S>> {
```

Comment:
Find the naming on this a bit awkward. It looks like the interface of an `Operation` (enables you to construct actions and obtain implied Operations). I can see that 

```
 Optional<Set<Class<? extends Operation<?>>>> supportedOperations();
 ```
might be harder to get at a glance that it's per resource, but that's the nature of an operation right, we would javadoc the Resource->Operation relationship.

Reply:
Okay, and modelling an additional Resource type in addition to Operation would add complexity for little gain.

With an enum per resource users just have to define a single enum to add their Resource.